### PR TITLE
vagrant: Upgrade Fedora 35 vagrantfile to Fedora 40

### PR DESCRIPTION
**Pull request by Haroon-Khel**
- [x] commit message has one of the [standard prefixes](https://github.com/adoptium/infrastructure/blob/master/CONTRIBUTING.md#commit-messages)
- [ ] [faq.md](https://github.com/adoptium/infrastructure/blob/master/FAQ.md) updated if appropriate
- [ ] other documentation is changed or added (if applicable)
- [x] playbook changes run through [VPC](https://ci.adoptium.net/view/Tooling/job/VagrantPlaybookCheck/) or [QPC](https://ci.adoptium.net/view/Tooling/job/QEMUPlaybookCheck/) (if you have access)
- [ ] VPC/QPC not applicable for this PR
- [ ] for inventory.yml changes, bastillion/nagios/jenkins updated accordingly

Fixes https://github.com/adoptium/infrastructure/issues/3662

### CODE REVIEW

**Comment by Haroon-Khel** [comment]
Ive temporarily changed the config of the VPC to pull from this branch. Running a fedora40 job https://ci.adoptium.net/view/Tooling/job/VagrantPlaybookCheck/1910/console

**Review by steelhead31 — COMMENTED** [review]
Looks OK, VPC fails, I think due to a similar issue I've had with Ubuntu 24.04.. if an older version of Python is available, you could try setting that up on the Fedora Box as part of the Vagrant file, and specifying the perl interpreter.

**Review by karianna — APPROVED** [review]
LGTM, feel free to merge post VPC success.

**Review by gdams — APPROVED** [review]

**Comment by Haroon-Khel** [comment]
Python2 seems to work but then I hit this error
```
19:09:50 TASK [logs : Update Log File] **************************************************
19:09:52 fatal: [127.0.0.1]: FAILED! => {"changed": false, "msg": "Aborting, target uses selinux but python bindings (libselinux-python) aren't installed!"}
```
The only available `libselinux-python` package on fedora40 is python3-libselinux which doesnt look like it helps if the default python is python2

File: ansible/vagrant/Vagrantfile.Fedora40 [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# -*- mode: ruby -*-
+# vi: set ft=ruby :
+
+$script = <<SCRIPT
+
+# Install python2, make it the default python
```
Comment (karianna) [inline ansible/vagrant/Vagrantfile.Fedora40:6]: Comment seems incorrect?

**Review by karianna — CHANGES_REQUESTED** [review]

**Comment by Haroon-Khel** [comment]
Eventhough `python3-libselinux` is installed, python3.8 is having trouble finding it

```
[root@fedora vagrant]# python3.8
Python 3.8.19 (default, Apr 17 2024, 00:00:00) 
[GCC 14.0.1 20240411 (Red Hat 14.0.1-0)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import selinux
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ModuleNotFoundError: No module named 'selinux'
>>> exit()
[root@fedora vagrant]# python3
Python 3.12.2 (main, Feb 21 2024, 00:00:00) [GCC 14.0.1 20240217 (Red Hat 14.0.1-0)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import selinux
>>>
```

**Comment by Haroon-Khel** [comment]
The wider solution is to update Ansible on the host, infrastructre-ibmcloud-vagrant-x64-1, so that it can use python3.12

**Comment by Haroon-Khel** [comment]
Installing `selinux` via pip looks promising

```
[root@fedora vagrant]# python3.8
Python 3.8.19 (default, Apr 17 2024, 00:00:00) 
[GCC 14.0.1 20240411 (Red Hat 14.0.1-0)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import selinux
>>> exit()
```

**Comment by Haroon-Khel** [comment]
Using `selinux` installed via pip and having `/usr/bin/python3` point to `/usr/bin/python3.8` I get as far as 

```
14:41:16 TASK [Common : YUM upgrade all packages] ***************************************
14:41:18 fatal: [127.0.0.1]: FAILED! => {"changed": false, "cmd": "dnf install -y python3-dnf", "msg": "Could not import the dnf python module using /usr/bin/python3.8 (3.8.19 (default, Apr 17 2024, 00:00:00) [GCC 14.0.1 20240411 (Red Hat 14.0.1-0)]). Please install `python3-dnf` package or ensure you have specified the correct ansible_python_interpreter.", "rc": 1, "results": [], "stderr": "Traceback (most recent call last):\n  File \"/usr/bin/dnf\", line 61, in <module>\n    from dnf.cli import main\nModuleNotFoundError: No module named 'dnf'\n", "stderr_lines": ["Traceback (most recent call last):", "  File \"/usr/bin/dnf\", line 61, in <module>", "    from dnf.cli import main", "ModuleNotFoundError: No module named 'dnf'"], "stdout": "", "stdout_lines": []}
14:41:18 
14:41:18 PLAY RECAP *********************************************************************
14:41:18 127.0.0.1                  : ok=13   changed=3    unreachable=0    failed=1    skipped=24   rescued=0    ignored=1   
```

**Review by karianna — APPROVED** [review]

**Comment by steelhead31** [comment]
Superceded by #3761

**Pull request closed without merging**